### PR TITLE
added make-admin feature

### PR DESCRIPTION
**Pull request by caseykim**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -15,4 +15,12 @@ def destroy
...
     redirect_to users_path
   end
+
+  def make_admin
+    @user = User.find(params[:user_id])
+    @user.role = "admin"
+    @user.save
+    flash[:success] = "#{@user.username} is now an admin."
+    redirect_to users_path
+  end
```
Comment (radavis) [inline app/controllers/users_controller.rb]: This method goes against Rails convention. You should update a user with the role admin in the params.

Comment (dreamingfurther) [inline app/controllers/users_controller.rb]: I agree this should be changed.

File: config/routes.rb [hunk 2]
```diff
@@ -2,7 +2,9 @@
   root 'tattoos#index'
   devise_for :users, controllers: { registrations: :registrations }
 
-  resources :users, only: [:index, :show, :destroy]
+  resources :users, only: [:index, :show, :destroy] do
+    post 'make_admin'
```
Comment (dreamingfurther) [inline config/routes.rb]: you should use a restful route for this action. Again just creating a custom route goes against Rails conventions. If you wanted for example, you could have a route:

```
resources :admin, only: [:create]
```

And that controller can update the user.

File: config/routes.rb [hunk 3]
```diff
@@ -2,7 +2,12 @@
   root 'tattoos#index'
   devise_for :users, controllers: { registrations: :registrations }
 
-  resources :users, only: [:index, :show, :destroy]
+  resources :users, only: [:index, :show, :destroy] do
+    # post 'make_admin'
```
Comment (dreamingfurther) [inline config/routes.rb]: remove the commented code

File: spec/features/admin_makes_user_admin_spec.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    visit users_path
+    find("#admin-#{user.id}", 'Make Admin').click
+
+    expect(page).to have_content("#{user.username} is now an admin.")
+    expect(User.find(id).role).to eq("admin")
+  end
+
+  scenario 'User cannot make another user admin' do
+    user = FactoryGirl.create(:user)
+    other_user = User.first
```
Comment (dreamingfurther) [inline spec/features/admin_makes_user_admin_spec.rb]: fix this hound violation

**Pull request merged**